### PR TITLE
[ Coverity ] Fix Coverity Issues

### PR DESCRIPTION
**Pull request by jijoongmoon**
This PR fix coverity issues.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: Applications/Classification/jni/main.cpp [hunk 1]
```diff
@@ -427,12 +427,23 @@ int main(int argc, char *argv[]) {
...
+    return 0;
+  }
   NN.readModel();
 
-  int status = NN.train();
+  try {
+    NN.train();
+  } catch (...) {
+    std::cerr << "Error during train" << std::endl;
+    return 0;
```
Comment (kparichay) [inline Applications/Classification/jni/main.cpp:442]: nn.finalize() before return??

Comment (kparichay) [inline Applications/Classification/jni/main.cpp:442]: nn.finalize() before return.

File: Applications/LogisticRegression/jni/main.cpp [hunk 2]
```diff
@@ -124,7 +124,25 @@ int main(int argc, char *argv[]) {
         std::vector<std::vector<float>> in, label;
         in.push_back(inputVector[j]);
         label.push_back(outputVector[j]);
-        NN.backwarding(nntrainer::Tensor(in), nntrainer::Tensor(label), i);
+        nntrainer::Tensor d, y;
+        try {
+          d = nntrainer::Tensor(in);
+        } catch (...) {
+          std::cerr << "Error during tensor construct" << std::endl;
+          return 0;
```
Comment (kparichay) [inline Applications/LogisticRegression/jni/main.cpp:132]: ditto.

File: Applications/LogisticRegression/jni/main.cpp [hunk 3]
```diff
@@ -124,7 +124,25 @@ int main(int argc, char *argv[]) {
...
+        }
+        try {
+          y = nntrainer::Tensor(label);
+        } catch (...) {
+          std::cerr << "Error during tensor construct" << std::endl;
+          return 0;
```
Comment (kparichay) [inline Applications/LogisticRegression/jni/main.cpp:138]: ditto

File: Applications/LogisticRegression/jni/main.cpp [hunk 4]
```diff
@@ -124,7 +124,25 @@ int main(int argc, char *argv[]) {
...
+        }
+        try {
+          NN.backwarding(d, y, i);
+        } catch (...) {
+          std::cerr << "Error during backwarding" << std::endl;
+          return 0;
```
Comment (kparichay) [inline Applications/LogisticRegression/jni/main.cpp:144]: ditto.

File: Applications/Training/jni/main.cpp [hunk 5]
```diff
@@ -256,8 +256,20 @@ int main(int argc, char *argv[]) {
...
+        in = nntrainer::Tensor({inputVector[j]});
+      } catch (...) {
+        std::cerr << "Error during tensor initialization" << std::endl;
+        return 0;
+      }
+      try {
+        out = nntrainer::Tensor({outputVector[j]});
+      } catch (...) {
+        std::cerr << "Error during tensor initialization" << std::endl;
+        return 0;
```
Comment (kparichay) [inline Applications/Training/jni/main.cpp:270]: ditto

File: nntrainer/src/neuralnet.cpp [hunk 6]
```diff
@@ -527,6 +527,11 @@ int NeuralNetwork::init() {
       status = layers[i]->setOptimizer(opt);
       NN_RETURN_STATUS();
 
+      if (i == 0) {
+        ml_loge("Batch normalization layer cannot be first layer of network");
```
Comment (kparichay) [inline nntrainer/src/neuralnet.cpp:531]: Is this check needed?

Comment (jijoongmoon) [reply nntrainer/src/neuralnet.cpp:531]: i could be 0 and i -1 is going to be negative.

File: Applications/Training/jni/main.cpp (hunk 5 above)
Comment (jijoongmoon) [reply Applications/Training/jni/main.cpp:270]: fixed.

File: Applications/LogisticRegression/jni/main.cpp (hunk 4 above)
Comment (jijoongmoon) [reply Applications/LogisticRegression/jni/main.cpp:144]: fixed.

File: Applications/LogisticRegression/jni/main.cpp (hunk 3 above)
Comment (jijoongmoon) [reply Applications/LogisticRegression/jni/main.cpp:138]: fixed.

File: Applications/LogisticRegression/jni/main.cpp (hunk 2 above)
Comment (jijoongmoon) [reply Applications/LogisticRegression/jni/main.cpp:132]: fixed.

File: Applications/Classification/jni/main.cpp (hunk 1 above)
Comment (jijoongmoon) [reply Applications/Classification/jni/main.cpp:442]: fixed.

**Pull request merged**